src/refmvs.rs

```diff
@@ -428,7 +430,7 @@ impl Rav1dRefmvsDSPContext {
         // For the Rust fallback fn, the extra args `&rf.r` and `ri`
         // are passed to allow for disjointedness checking.
         let rr = &ri.map(|ri| {
-            if ri > rf.r.len() {
+            if ri > rf.r.len() - R_PAD {
                 return ptr::null();
             }
             // SAFETY: `.add` is in-bounds; checked above.
```

Comment:
```suggestion
            // SAFETY: `.add` is in-bounds; checked above.
            // Also note that asm may read 12-byte `refmvs_block`s in 16-byte chunks.
            // This is safe because we allocate `rf.r` with an extra `R_PAD` (1) elements.
            // These ptrs are only read, so these overlapping reads are safe
            // (only read is only checked in the fallback Rust `fn`).
            // Furthermore, this is provenance safe because
            // we derive the ptrs from `rf.r.as_mut_ptr()`,
            // as opposed to materializing intermediate references.
            const _: () = assert!(mem::size_of::<refmvs_block>() * (1 + R_PAD) > 16);
```